core/src/test/java/org/testcontainers/utility/TestcontainersConfigurationTest.java

```diff
@@ -1,57 +1,112 @@
+        assertEquals(
+            "an image name can be pulled from classpath properties",
+            DockerImageName.parse("foo:version"),
+            newConfig().getConfiguredSubstituteImage(DockerImageName.parse("testcontainers/ryuk:any"))
+        );
+    }
+
+    @Test
+    public void shouldSubstituteImageNamesFromUserProperties() {
+        userProperties.setProperty("ryuk.container.image", "foo:version");
+        assertEquals(
+            "an image name can be pulled from user properties",
+            DockerImageName.parse("foo:version"),
+            newConfig().getConfiguredSubstituteImage(DockerImageName.parse("testcontainers/ryuk:any"))
+        );
+    }
+
+    @Test
+    public void shouldSubstituteImageNamesFromEnvironmentVariables() {
+        environment.put("TESTCONTAINERS_RYUK_CONTAINER_IMAGE", "foo:version");
+        assertEquals(
+            "an image name can be pulled from an environment variable",
+            DockerImageName.parse("foo:version"),
+            newConfig().getConfiguredSubstituteImage(DockerImageName.parse("testcontainers/ryuk:any"))
+        );
+    }
+
+    @Test
+    public void shouldReadChecksFromUserPropertiesOrEnvironmentOnly() {
         assertFalse("checks enabled by default", newConfig().isDisableChecks());
 
         classpathProperties.setProperty("checks.disable", "true");
         assertFalse("checks are not affected by classpath properties", newConfig().isDisableChecks());
 
-        environmentProperties.setProperty("checks.disable", "true");
-        assertTrue("checks disabled", newConfig().isDisableChecks());
+        userProperties.setProperty("checks.disable", "true");
+        assertTrue("checks disabled via user properties", newConfig().isDisableChecks());
+
+        userProperties.remove("checks.disable");
+        environment.put("TESTCONTAINERS_CHECKS_DISABLE", "true");
+        assertTrue("checks disabled via env var", newConfig().isDisableChecks());
     }
 
     @Test
-    public void shouldReadDockerClientStrategyFromEnvironmentOnly() {
+    public void shouldReadDockerClientStrategyFromUserPropertiesOrEnvironmentOnly() {
         String currentValue = newConfig().getDockerClientStrategyClassName();
 
         classpathProperties.setProperty("docker.client.strategy", UUID.randomUUID().toString());
         assertEquals("Docker client strategy is not affected by classpath properties", currentValue, newConfig().getDockerClientStrategyClassName());
 
-        environmentProperties.setProperty("docker.client.strategy", "foo");
-        assertEquals("Docker client strategy is changed", "foo", newConfig().getDockerClientStrategyClassName());
+        userProperties.setProperty("docker.client.strategy", "foo");
+        assertEquals("Docker client strategy is changed by user property", "foo", newConfig().getDockerClientStrategyClassName());
+
+        userProperties.remove("docker.client.strategy");
```

Comment:
WDYT about either adding an extra assert after the property is removed (to ensure that it is not cached, for example) or splitting the test into 2 (user properties, environment)?

Reply:
Done